hathor/transaction/storage/transaction_storage.py

```diff
@@ -458,7 +461,43 @@ def is_empty(self) -> bool:
 
     def pre_init(self) -> None:
         """Storages can implement this to run code before transaction loading starts"""
-        pass
+        self._check_and_set_network()
+
+    def _check_and_set_network(self) -> None:
+        from hathor.transaction.storage.exceptions import WrongNetworkError
+        network = settings.NETWORK_NAME
+        stored_network = self.get_network()
+        if stored_network is None:
```

Comment:
The logic seems great. I don't like these nested if statements because they are hard to read. Not sure if we can refactor for an easier reading.

Reply:
What if I split out the inside of `if stored_network is None` into a `def _try_to_infer_network(self)` method?

Reply:
Refactored.

Reply:
Great! Thanks!